src/hal/src/queue/mod.rs

```diff
@@ -104,6 +158,38 @@ pub trait CommandQueue<B: Backend>: fmt::Debug + Any + Send + Sync {
         Iw: IntoIterator<Item = (&'a S, pso::PipelineStage)>,
         Is: IntoIterator<Item = &'a S>;
 
+    /// Sparse memory bind operation.
+    ///
+    /// # Arguments
+    ///
+    /// * `info` - information about the memory bindings.
+    ///
+    /// # Safety
+    ///
+    /// - Defining memory as `None` will cause undefined behaviour when the
+    /// tile is read or written from in some hardware.
+    /// - The memory regions provided are not checked to be valid and matching
+    /// of the sparse resource type.
+    /// - If extents are not a multiple of the block size, additional space will be
+    /// bound, and accessing memory is unsafe.
+    unsafe fn bind_sparse<'a, M, Bf, I, S, Iw, Is, Ibi, Ib, Iii, Io, Ii>(
+        &mut self,
+        info: BindSparseInfo<Iw, Is, Ib, Io, Ii>,
+        device: &B::Device,
+        fence: Option<&B::Fence>,
+    ) where
+        Bf: 'a + BorrowMut<B::Buffer>,
+        M: 'a + Borrow<B::Memory>,
+        Ibi: IntoIterator<Item = SparseMemoryBind<&'a M>>,
```

Comment:
In other entry points we also have the `ExactSizeIterator` bound (e.g. see `create_render_pass`), which allows the backends to allocate stuff on stack instead of the heap. Let's use it here?